zomes/rea_plan/lib/src/lib.rs

```diff
@@ -0,0 +1,93 @@
+/**
+ * Holo-REA plan zome library API
+ *
+ * Contains helper methods that can be used to manipulate `Plan` data
+ * structures in either the local Holochain zome, or a separate DNA-local zome.
+ *
+ * @package Holo-REA
+ */
+use paste::paste;
+use hdk_records::{
+    RecordAPIResult,
+    records::{
+        create_record,
+        read_record_entry,
+        update_record,
+        delete_record,
+    },
+};
+use hdk_semantic_indexes_client_lib::*;
+
+use hc_zome_rea_plan_storage::*;
+use hc_zome_rea_plan_rpc::*;
+
+pub use hc_zome_rea_plan_storage::PLAN_ENTRY_TYPE;
+
+pub fn handle_create_plan<S>(entry_def_id: S, plan: CreateRequest) -> RecordAPIResult<ResponseData>
+    where S: AsRef<str>
+{
+    let (header_addr, base_address, entry_resp): (_,_, EntryData) = create_record(&entry_def_id, plan)?;
+    construct_response(&base_address, header_addr, &entry_resp, get_link_fields(&base_address)?)
+}
+
+pub fn handle_get_plan<S>(entry_def_id: S, address: PlanAddress) -> RecordAPIResult<ResponseData>
+    where S: AsRef<str>
+{
+    let (revision, base_address, entry) = read_record_entry::<EntryData, EntryStorage, _,_,_>(&entry_def_id, address.as_ref())?;
+    construct_response(&base_address, revision, &entry, get_link_fields(&base_address)?)
+}
+
+pub fn handle_update_plan<S>(entry_def_id: S, plan: UpdateRequest) -> RecordAPIResult<ResponseData>
+    where S: AsRef<str>
+{
+    let revision_hash = plan.get_revision_id().clone();
+    let (revision_id, identity_address, entry, _prev_entry): (_,_, EntryData, EntryData) = update_record(&entry_def_id, &revision_hash, plan)?;
+    construct_response(&identity_address, revision_id, &entry, get_link_fields(&identity_address)?)
+}
+
+pub fn handle_delete_plan(address: HeaderHash) -> RecordAPIResult<bool> {
+    delete_record::<EntryData>(&address)
+}
+
+/// Create response from input DHT primitives
+fn construct_response<'a>(
+    address: &PlanAddress, revision: HeaderHash, e: &EntryData, (
+        processes,
+        commitments,
```

Comment:
In retrospect, I am thinking that this variable could and should be named
`independent_demands`
for consistency.